### PR TITLE
Allow blocking by source address on Port 43 Queries

### PR DESCRIPTION
**Pull request by maggarwal13**
The [RIPE Database Acceptable Use Policy]() includes the option to block access:
```
The RIPE Database Terms and Conditions permit the RIPE NCC to block 
Access to the RIPE Database services by any User for abuse, 
or suspend Access for suspected abuse.
```

If this is necessary then allow blocking by source address on port 43 queries.

### CODE REVIEW

File: whois-api/src/main/java/net/ripe/db/whois/api/httpserver/IpBlockListFilter.java [hunk 1]
```diff
@@ -37,7 +36,7 @@ public void init(FilterConfig filterConfig) throws ServletException {}
     public void doFilter(final ServletRequest servletRequest, final ServletResponse servletResponse, final FilterChain filterChain) throws IOException, ServletException {
         final HttpServletRequest httpRequest = (HttpServletRequest) servletRequest;
 
-        if (isBlockedIp(httpRequest.getRemoteAddr())){
+        if (ipBlockManager.isBlockedIp(InetAddresses.forString(httpRequest.getRemoteAddr()))){
```
Comment (eshryane) [inline whois-api/src/main/java/net/ripe/db/whois/api/httpserver/IpBlockListFilter.java:39]: Add an overloaded method `isBlockedIp` to accommodate `request.getRemoteAddress()` so you can handle the conversion within `IpBlockManager`

File: whois-api/src/main/java/net/ripe/db/whois/api/httpserver/IpBlockListFilter.java [hunk 2]
```diff
@@ -50,22 +49,9 @@ public void destroy() {
...
-        return false;
-    }
-
     private static void sendError(final HttpServletResponse httpResponse, final HttpServletRequest httpRequest) throws IOException {
         //TODO: refactor using Response.writeError() when upgrading to Jetty 12 https://jetty.org/docs/jetty/12/programming-guide/migration/11-to-12.html
-        final String message = String.format("Your host %s has been permanently blocked due to suspected abusive " +
-                "behaviour. Please contact support for further assistance.", httpRequest.getRemoteAddr());
         httpResponse.setStatus(HttpStatus.TOO_MANY_REQUESTS_429);
-        httpResponse.getWriter().write(message);
+        httpResponse.getWriter().write(QueryMessages.accessDeniedForAbuse(httpRequest.getRemoteAddr()).getFormattedText());
```
Comment (eshryane) [inline whois-api/src/main/java/net/ripe/db/whois/api/httpserver/IpBlockListFilter.java:55]: Much better! Thanks

File: whois-client/src/main/java/net/ripe/db/whois/query/QueryMessages.java [hunk 3]
```diff
@@ -291,6 +291,12 @@ public static Message accessDeniedPermanently(final String accountingId) {
                 accountingId);
     }
 
+    public static Message accessDeniedForAbuse(final String accountingId) {
+        return new QueryMessage(Type.ERROR, ""
+                + "Your host %s has been permanently blocked due to suspected abusive behaviour. Please contact support for further assistance.",
```
Comment (eshryane) [inline whois-client/src/main/java/net/ripe/db/whois/query/QueryMessages.java:296]: Include an email address `<ripe-dbm@ripe.net>` after `support`.

Comment (eshryane) [reply whois-client/src/main/java/net/ripe/db/whois/query/QueryMessages.java:296]: We also should review the text and maybe replace `suspected abusive` with something more neutral, as they may be blocked inadvertently

File: whois-commons/src/main/java/net/ripe/db/whois/common/hazelcast/IpBlockManager.java [hunk 4]
```diff
@@ -24,4 +26,6 @@ default List<? extends IpInterval<?>> getBlockedIntervals(String blockedListIps)
         }
         return Stream.of(blockedListIps.split(",")).map(IpInterval::parse).toList();
     }
+
+    boolean isBlockedIp(final InetAddress candidate);
```
Comment (eshryane) [inline whois-commons/src/main/java/net/ripe/db/whois/common/hazelcast/IpBlockManager.java:30]: Add `isBlockedIp(String)`

**Review by eshryane — COMMENTED** [review]
Small comments, approved otherwise, thanks

File: whois-client/src/main/java/net/ripe/db/whois/query/QueryMessages.java (hunk 3 above)
Comment (maggarwal13) [reply whois-client/src/main/java/net/ripe/db/whois/query/QueryMessages.java:296]: Decided to use "Your host %s has been blocked. Please contact support <ripe-dbm@ripe.net> for further assistance."

**Review by eshryane — APPROVED** [review]

**Pull request merged**